Essentials/src/main/java/com/earth2me/essentials/commands/Commandpay.java

```diff
@@ -46,28 +46,32 @@ public void run(final Server server, final User user, final String commandLabel,
         }
 
         BigDecimal tempAmount = new BigDecimal(sanitizedString);
-        switch (Character.toLowerCase(ogStr.charAt(ogStr.length() - 1))) {
-            case 'k': {
-                tempAmount = tempAmount.multiply(THOUSAND);
-                break;
-            }
-            case 'm': {
-                tempAmount = tempAmount.multiply(MILLION);
-                break;
-            }
-            case 'b': {
-                tempAmount = tempAmount.multiply(BILLION);
-                break;
-            }
-            case 't': {
-                tempAmount = tempAmount.multiply(TRILLION);
-                break;
-            }
-            default: {
-                break;
+        if (ess.getSettings().isPayModifierEnabled()) {
+            final String modifier = ogStr.toLowerCase().replaceAll("[^a-z]", "");
+            if (!modifier.isEmpty()) {
+                switch (modifier) {
+                    case "k": {
+                        tempAmount = tempAmount.multiply(THOUSAND);
+                        break;
+                    }
+                    case "m": {
+                        tempAmount = tempAmount.multiply(MILLION);
+                        break;
+                    }
+                    case "b": {
+                        tempAmount = tempAmount.multiply(BILLION);
+                        break;
+                    }
+                    case "t": {
+                        tempAmount = tempAmount.multiply(TRILLION);
+                        break;
+                    }
+                    default: {
+                        throw new InvalidModifierException();
+                    }
+                }
             }
         }
-
```

Comment:
Try not to change formatting if possible such as adding/removing new lines in areas not related to your PR

Reply:
This is essentials for my pr to work